plugin/storage/badger/factory_test.go

```diff
@@ -144,3 +144,33 @@ func TestMaintenanceCodecov(t *testing.T) {
 	_ = f.store.Close()
 	waiter() // This should trigger the logging of error
 }
+
+func TestBadgerMetrics(t *testing.T) {
+	f := NewFactory()
+	v, command := config.Viperize(f.AddFlags)
+	command.ParseFlags([]string{
+		"--badger.metrics-update-interval=10ms",
+	})
+	f.InitFromViper(v)
+	mFactory := metricstest.NewFactory(0)
+	f.Initialize(mFactory, zap.NewNop())
+	assert.NotNil(t, f.metrics.badgerMetrics)
+	_, found := f.metrics.badgerMetrics["badger_memtable_gets_total"]
+	assert.True(t, found)
+
+	waiter := func(previousValue int64) int64 {
+		sleeps := 0
+		_, gs := mFactory.Snapshot()
+		for gs["adger_memtable_gets_total"] == previousValue && sleeps < 8 {
```

Comment:
why "adger"?

Reply:
Typo. It always sleeps the maximum with that typo, so the test functions correctly - but might take too long. I'll fix that.